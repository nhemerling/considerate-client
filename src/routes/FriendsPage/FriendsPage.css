.FriendsPage__friends {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
}

.FriendsPage input {
  width: 100%;
}

@media screen and (min-width: 450px) {
  .FriendsPage input {
    width: 50%;
  }
}
