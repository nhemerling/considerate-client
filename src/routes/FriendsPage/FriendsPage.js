import React, { Component } from 'react';
import FriendCard from '../../components/FriendCard/FriendCard';
import FriendApiService from '../../services/friend-api-service';
import ConsiderateContext from '../../context/ConsiderateContext';
import './FriendsPage.css';

export default class FriendsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      error: null,
    };
  }
  static contextType = ConsiderateContext;

  componentDidMount() {
    FriendApiService.getFriends()
      .then(this.context.setFriendList)
      .catch(this.context.setError);
  }

  updateSearch(event) {
    this.setState({
      search: event.target.value,
    });
  }

  compare(a, b) {
    const dateA = a.occasion_date;
    const dateB = b.occasion_date;

    let comparison = 0;
    if (dateA > dateB) {
      comparison = 1;
    } else if (dateA < dateB) {
      comparison = -1;
    }
    return comparison;
  }

  renderFilteredFriends() {
    const { friendList = [] } = this.context;
    const sortedFriends = friendList.sort(this.compare);
    const filteredFriends = sortedFriends.filter((friend) => {
      return (
        friend.friend_name
          .toLowerCase()
          .indexOf(this.state.search.toLowerCase()) !== -1
      );
    });
    return filteredFriends.map((friend) => (
      <FriendCard key={friend.id} friend={friend} />
    ));
  }

  render() {
    return (
      <section className="HomePage">
        <label htmlFor="search">Search: </label>
        <input
          type="text"
          name="search"
          value={this.state.search}
          onChange={(e) => this.updateSearch(e)}
        ></input>
        {this.renderFilteredFriends()}
      </section>
    );
  }
}
