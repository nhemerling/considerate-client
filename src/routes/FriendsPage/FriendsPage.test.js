import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FriendsPage from './FriendsPage';

describe('FriendsPage component', () => {
  it('renders a FriendsPage component by default', () => {
    const wrapper = shallow(<FriendsPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
