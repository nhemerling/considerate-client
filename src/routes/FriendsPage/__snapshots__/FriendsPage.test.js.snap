// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsPage component renders a FriendsPage component by default 1`] = `
<section
  className="FriendsPage"
>
  <label
    htmlFor="search"
  >
    <p>
      Search Friends: 
    </p>
  </label>
  <input
    name="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="FriendsPage__friends"
  />
</section>
`;
